core/main/src/firebolt/handlers/capabilities_rpc.rs

```diff
@@ -304,16 +302,16 @@ pub async fn is_permitted(
         .check_cap_role(&ctx.app_id, cap.clone())
     {
         return Ok(v);
-    } else {
-        if let Ok(_) = PermissionHandler::fetch_and_store(state.clone(), ctx.clone().app_id).await {
-            //successful fetch retry
-            if let Ok(v) = state
-                .cap_state
-                .permitted_state
-                .check_cap_role(&ctx.app_id, cap)
-            {
-                return Ok(v);
-            }
+    } else if let Ok(_) =
+        PermissionHandler::fetch_and_store(state.clone(), ctx.clone().app_id).await
```

Comment:
If we change the function signature we can avoid the second clone here as well. In fact, we can even make the function take reference for platform state and app id so that we can avoid both clones in this function.